File: 3245-find-beautiful-indices-in-the-given-array-i/find-beautiful-indices-in-the-given-array-i.py
class Solution:
    def beautifulIndices(self, s: str, a: str, b: str, k: int) -> List[int]:
        # [i...i+a.len] == a, then i is a candidate
        # [j...j+b.len] == b, then j is b candidate

        n = len(s)
        na = len(a)
        nb = len(b)

        beautList = []
        aIndexList = []
        bIndexList = []

        for i in range(n-na+1):
            curr_a = s[i:i+na]
            if curr_a==a:
                aIndexList.append(i)
        
        for j in range(n-nb+1):
            curr_b = s[j:j+nb]
            if curr_b==b:
                bIndexList.append(j)

        # print(aIndexList, bIndexList)
        for i in aIndexList:
            for j in bIndexList:
                if abs(j-i)<= k:
                    beautList.append(i)
                    break
        return beautList

File: 3245-find-beautiful-indices-in-the-given-array-i/find-beautiful-indices-in-the-given-array-i.java
class Solution {
    public List<Integer> beautifulIndices(String s, String a, String b, int k) {
        
        int n = s.length();
        int na = a.length();
        int nb = b.length();
        List<Integer> res = new ArrayList<>();

        List<Integer> aIdx = new ArrayList<>();
        List<Integer> bIdx = new ArrayList<>();


        for(Integer i=0; i < n-na+1; i++) {
            String currA = s.substring(i, i+na);
            if (currA.equals(a)) {
                aIdx.add(i);
            }
        }

        for(Integer j=0; j < n-nb+1; j++) {
            String currB = s.substring(j, j+nb);
            if (currB.equals(b)) {
                bIdx.add(j);
            }
        }

        for(Integer i: aIdx){
            for(Integer j: bIdx) {
                if(Math.abs(j-i) <= k) {
                    res.add(i);
                    break;
                }
            }
        }
        
       return res; 
    }
}